test/parallel/test-util-format.js

```diff
@@ -57,20 +57,20 @@ assert.throws(
   }
 );
 
-// Number format specifier
+// Integer format specifier
 assert.strictEqual(util.format('%d'), '%d');
 assert.strictEqual(util.format('%d', 42.0), '42');
 assert.strictEqual(util.format('%d', 42), '42');
 assert.strictEqual(util.format('%d', '42'), '42');
 assert.strictEqual(util.format('%d', '42.0'), '42');
-assert.strictEqual(util.format('%d', 1.5), '1.5');
-assert.strictEqual(util.format('%d', -0.5), '-0.5');
-assert.strictEqual(util.format('%d', ''), '0');
+assert.strictEqual(util.format('%d', 1.5), '1');
+assert.strictEqual(util.format('%d', -0.5), '0');
+assert.strictEqual(util.format('%d', ''), 'NaN');
 assert.strictEqual(util.format('%d %d', 42, 43), '42 43');
 assert.strictEqual(util.format('%d %d', 42), '42 %d');
 assert.strictEqual(
   util.format('%d', 1180591620717411303424),
-  '1.1805916207174113e+21'
+  '1'
```

Comment:
I’m a bit surprised by this one, tbh … at least for me locally, `` `${parseInt('1180591620717411303424')}` === '1.1805916207174113e+21' ``? Do you know why there’s a discrepancy?

Interestingly, Firefox seems to provide `0` here, and Chrome the exponential notation…

Reply:
What happens here is not ``${parseInt('1180591620717411303424')}` but `${parseInt(1180591620717411303424)}` which is equivalent to `${parseInt('1.1805916207174113e+21')}` and returns 1.

Reply:
So … just to make sure, these are browser bugs?

Reply:
Not sure... It could also be a spec bug because no browser has the "correct behavior.
To summarize, with `console.log('%d', 1180591620717411303424)`:
- Edge and Chrome: `1.1805916207174113e+21`
- Firefox: `0`
- Safari: TODO (anyone on a mac please update)

Reply:
To me it seems like the browsers deviate from the spec in this case. However, it does seem to be a a good idea to follow Chrome and Edge in this case. Especially, since the spec does not seem to follow any browser implementation in this case.

Reply:
Safari logs `1`

Reply:
`1` should be correct according to the [latest parseInt spec](https://tc39.github.io/ecma262/#sec-parseint-string-radix). `NumberToString` has to convert the number to expotential notation and then the following spec text applies:

> If S contains a code unit that is not a radix-R digit, let Z be the substring of S consisting of all code units before the first such code unit

> Let mathInt be the mathematical integer value that is represented by Z

`.` is not a radix-R digit, so parsing stops there.

Reply:
So in summary for `console.log('%d', number)`:

- Chrome uses `Number`
- Edge uses `Number`
- Safari uses `parseInt` (this is spec-compliant)
- Firefox uses neither `Number` nor `parseInt`

Reply:
Chrome may not use `Number` for `%d`.
at least chrome 70(stable) on my macbook, I think `%d` is not equal `Number`.
the results are:
`console.log('%d', 1.2)`: `1`
`Number(1.2)`: `1.2`